scripts/zones/AlTaieu/mobs/Jailer_of_Love.lua

```diff
@@ -20,76 +22,283 @@ local minionGroup =
+    local phuaboUp = {}
+    local phuaboDn = {}
+    for i = ID.mob.JAILER_OF_LOVE + 1, ID.mob.JAILER_OF_LOVE + 9 do
+        local phuabo = GetMobByID(i)
+        if phuabo:isAlive() then
+            table.insert(phuaboUp, i)
+        elseif not phuabo:isSpawned() then
+            table.insert(phuaboDn, i)
+        end
+    end
+
+    -- how many sharks spawn depends on number currently alive
+    -- https://www.bg-wiki.com/bg/Jailer_of_Love
+    local numToSpawn = 0
+    if #phuaboUp == 2 then
+        numToSpawn = 1
+    elseif #phuaboUp == 0 or #phuaboUp == 3 then
+        numToSpawn = 3
+    elseif #phuaboUp == 1 then
+        numToSpawn = math.random(3)
+    end
+
+    -- spawn sharks
+    for i = 1, math.min(numToSpawn, #phuaboDn) do
+        -- spawnSharks(mob, phuaboDn)
+        local target = mob:getTarget()
+        local phuabo = GetMobByID(phuaboDn[i])
+
+        if phuabo then
+            phuabo:setSpawn(target:getXPos() + math.random(-2, 2), target:getYPos(), target:getZPos())
+            SpawnMob(phuaboDn[i])
+            phuabo:setMobMod(xi.mobMod.SUPERLINK, mob:getTargID())
+            phuabo:updateEnmity(target)
+        end
+    end
+end
+
+entity.onMobInitialize = function(mob)
+    mob:setBehaviour(2)
+    mob:setMagicCastingEnabled(true)
+end
+
+entity.onMobSpawn = function(mob)
+    mob:addMod(xi.mod.REGEN, 260)
+    mob:setMod(xi.mod.DMGMAGIC, -5000) -- starts the fight with -50% magic damage taken, reduced to 25% after regen is taken off.
+    mob:setMod(xi.mod.ATT, 452)
+    mob:setMod(xi.mod.DEF, 620)
+    mob:setMod(xi.mod.EVA, 328)
+
+    mob:addImmunity(xi.immunity.SLEEP)
+    mob:addImmunity(xi.immunity.GRAVITY)
+    mob:addImmunity(xi.immunity.BIND)
+    mob:addImmunity(xi.immunity.STUN)
+    mob:addImmunity(xi.immunity.SILENCE)
+    mob:addImmunity(xi.immunity.PARALYZE)
+    mob:addImmunity(xi.immunity.BLIND)
+    mob:addImmunity(xi.immunity.SLOW)
+    mob:addImmunity(xi.immunity.POISON)
+    mob:addImmunity(xi.immunity.ELEGY)
+    mob:addImmunity(xi.immunity.REQUIEM)
+    mob:addImmunity(xi.immunity.TERROR)
+
+    xi.mix.jobSpecial.config(mob, {
+        specials =
+        {
+            { id = xi.jsa.ASTRAL_FLOW, hpp = math.random(45, 55) },
+        },
+    })
+end
+
+entity.onMobEngaged = function(mob, target)
     mob:hideName(false)
     mob:setUntargetable(false)
+    mob:setMagicCastingEnabled(true)
     mob:setAnimationSub(2)
+    mob:setLocalVar('elementAbsorb', os.time() + 120)
+    mob:setLocalVar('pop_pets', os.time() + 150) -- wait 2.5 minutes until spawning initial mobs
+    mob:setLocalVar('currentAbsorb', math.random(#eleAbsorbModID)) -- pick a random element to absorb after engaging
+    mob:setSpellList(spellLists[mob:getLocalVar('currentAbsorb')])
+    mob:setMod(eleAbsorbModID[mob:getLocalVar('currentAbsorb')], 100)
 end
 
 entity.onMobFight = function(mob, target)
-    -- reduce regen after nine Xzomits and Hpemdes are killed
+    mob:setAnimationSub(2)
```

Comment:
This seems weird. This would mean we are constantly setting its animationSub. We are already doing that onMobSpawn and I dont see it being changed anywhere else

Reply:
Commented for now, but leaving in as a note for myself when I test